docs/esp32/quickref.rst

```diff
@@ -240,6 +240,73 @@ ESP32 specific ADC class method reference:
       - ``ADC.WIDTH_11BIT``: 11 bit data
       - ``ADC.WIDTH_12BIT``: 12 bit data - this is the default configuration
 
+DAC (digital to analog conversion)
+----------------------------------
+
+On the ESP32 DAC functionality is available on Pins 25 and 26. Note that, when creating
+the DAC object for a pin, the DAC value will be initialized at 0.
+
+Use the :ref:`machine.DAC <machine.DAC>` class::
+
+    from machine import DAC
+
+    dac = DAC(Pin(25))          # create DAC object on pin 25
+    dac.write(128)              # set the output voltage to 1.65V
+
+    dac.cosine_enable()         # enable the cosine generator
+    dac.frequency_set(7, 1)     # set the output frequency using the RTC clock divider and 
+                                # frequency steps for the CW generator, (7, 1) -> ~15 Hz
+    
+
+ESP32 specific DAC class method reference:
+
+.. method:: DAC.cosine_enable()
+
+    This method enables the cosine generator for the DAC.
+
+.. method:: DAC.cosine_disable()
+
+    This method disables the cosine generator for the DAC.
+
+.. method:: DAC.frequency_step(frequency_step)
+
+    This method sets the frequency steps of the CW generator. The base is the 8 MHz RTC clock. The 
+    ``frequency_step`` adjusts the frequency as ``rtc_clock x frequency_step / 65536``. The RTC
+    clock is by default 8 MHz, and can be divided down by ``DAC.rtc_clk_div``.
+
+.. method:: DAC.rtc_clk_div(clk_8m_div)
+    This method sets the RTC clock divider. This allows to achieve lower frequencies of the CW generator.
+    ``clk_8m_div`` is the divider and must be between 0 and 7.
+
+.. Warning::
+    Be cautious in changing this value as it might affect other peripherals that use the 8 MHz RTC clock!
```

Comment:
Should this method be in a different class if it affects the system globally? E.g. `ESP32.rtc_clk_div`? I know it's more work, but it would be awful to have 2-3 peripherals that each have an `rtc_clk_div` method and that conflict with one another because they really adjust the same thing.

Reply:
Agree. Will move that over and make a note about it in the reference for the frequency set function